examples/platform/silabs/efr32/rs911x/wfx_rsi.h

```diff
@@ -1,106 +0,0 @@
-/*
- *
- *    Copyright (c) 2022 Project CHIP Authors
- *
- *    Licensed under the Apache License, Version 2.0 (the "License");
- *    you may not use this file except in compliance with the License.
- *    You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- *    Unless required by applicable law or agreed to in writing, software
- *    distributed under the License is distributed on an "AS IS" BASIS,
- *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *    See the License for the specific language governing permissions and
- *    limitations under the License.
- */
-
-#ifndef _WFX_RSI_H_
-#define _WFX_RSI_H_
-/*
- * Interface to RSI Sapis
- */
-
-#define WFX_RSI_WLAN_TASK_SZ (1024 + 512 + 256) /* Unknown how big this should be 	*/
-#define WFX_RSI_TASK_SZ (1024 + 1024)           /* Stack for the WFX/RSI task		*/
-#define WFX_RSI_BUF_SZ (1024 * 10)              /* May need tweak 			*/
-#define WFX_RSI_CONFIG_MAX_JOIN (5)             /* Max join retries			*/
-#define WFX_RSI_NUM_TIMERS (2)                  /* Number of RSI timers to alloc	*/
-
-/*
- * Various events fielded by the wfx_rsi task
- * Make sure that we only use 8 bits (otherwise freeRTOS - may need some changes)
- */
-#define WFX_EVT_STA_CONN (0x01)
-#define WFX_EVT_STA_DISCONN (0x02)
-#define WFX_EVT_AP_START (0x04)
-#define WFX_EVT_AP_STOP (0x08)
-#define WFX_EVT_SCAN (0x10) /* This is used as scan result and start */
-#define WFX_EVT_STA_START_JOIN (0x20)
-#define WFX_EVT_STA_DO_DHCP (0x40)
-#define WFX_EVT_STA_DHCP_DONE (0x80)
-
-#define WFX_RSI_ST_DEV_READY (0x01)
-#define WFX_RSI_ST_AP_READY (0x02)
-#define WFX_RSI_ST_STA_PROVISIONED (0x04)
-#define WFX_RSI_ST_STA_CONNECTING (0x08)
-#define WFX_RSI_ST_STA_CONNECTED (0x10)
-#define WFX_RSI_ST_STA_DHCP_DONE (0x40) /* Requested to do DHCP after conn	*/
-#define WFX_RSI_ST_STA_MODE (0x80)      /* Enable Station Mode			*/
-#define WFX_RSI_ST_AP_MODE (0x100)      /* Enable AP Mode			*/
-#define WFX_RSI_ST_STA_READY (WFX_RSI_ST_STA_CONNECTED | WFX_RSI_ST_STA_DHCP_DONE)
-#define WFX_RSI_ST_STARTED (0x200)     /* RSI task started			*/
-#define WFX_RSI_ST_SCANSTARTED (0x400) /* Scan Started				*/
-#define WFX_RSI_ST_SLEEP_READY (0x800) /* Notify the M4 to go to sleep*/
-
-typedef struct wfx_rsi_s
```

Comment:
This struct is different between the 917 and and the rs911x, rsi_if still uses wfx_rsi.events, which can otherwise be removed in SiWx917/wfx_rsi.h.

Reply:
Unfortunately, both platform share the sl_wifi_if.cpp file (through the symbolic link), which means any changes done on sl_wifi_if.cpp should compile in RS911 as well. Without the shared header, the RS911 won't compile.